src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php

```diff
@@ -281,7 +280,30 @@ protected function describeContainerParameter($parameter, array $options = [])
 
     protected function describeContainerEnvVars(array $envs, array $options = [])
     {
-        throw new LogicException('Using the markdown format to debug environment variables is not supported.');
+        $this->write("# Environment Variables\n\n");
+
+        if ([] === $envs) {
+            $this->write("There are no environment variables\n");
+
+            return;
+        }
+
+        $this->write("NAME | PROCESSOR | DEFAULT | VALUE\n");
+        $this->write("---|---|---|---\n");
```

Comment:
Would it be more readable if the table was pretty-printed with aligned pipes?

Reply:
I wouldn't care much. All methods in that class are hard to read because of that markdown string manipulation things. I would prefer that syntax, because it's the default markdown for a table header